Sync `b` and `strong` to `font-weight: bolder` as per Web Specification

Ahmad-S792 opened the pull request:
#### 2f7d04cbae67d35d6c3f2830cb134313c9b0dde1
<pre>
Sync `b` and `strong` to `font-weight: bolder` as per Web Specification

<a href="https://bugs.webkit.org/show_bug.cgi?id=289297">https://bugs.webkit.org/show_bug.cgi?id=289297</a>
<a href="https://rdar.apple.com/146458131">rdar://146458131</a>

Reviewed by Tim Nguyen.

This patch aligns WebKit with Web Specification [1]:

[1] <a href="https://html.spec.whatwg.org/multipage/rendering.html#phrasing-content-3">https://html.spec.whatwg.org/multipage/rendering.html#phrasing-content-3</a>

Our current implementation had `font-weight: bold` for `strong` and `b` and
this patch fixes it to make it `font-weight: bolder` and align with other
browser engines as well.

It also updates local test similar to below Blink Commit [2]:

[2] <a href="https://chromium-review.googlesource.com/c/chromium/src/+/6269883">https://chromium-review.googlesource.com/c/chromium/src/+/6269883</a>

From this change, we start hitting `primitiveValue-&gt;isNumber()` assertion and
looking into `fontWeightValueIsBold` function, we were hitting this assertion because
in `switch` statement, we were not handling `CSSValueBolder`, so `bolder` was
hitting assertion, which is to ensure that it is `number` only.

* LayoutTests/TestExpectations:
* Source/WebCore/css/html.css:
(strong, b):
* Source/WebCore/editing/EditingStyle.cpp:
(fontWeightValueIsBold):

&gt; Renamed `font-weight-bold-for-b-and-strong.html` to `font-weight-bolder-for-b-and-strong.html`:
* LayoutTests/platform/gtk/fast/html/font-weight-bold-for-b-and-strong-expected.png: Removed.
* LayoutTests/platform/ios/fast/html/font-weight-bold-for-b-and-strong-expected.txt: Ditto
* LayoutTests/platform/mac/fast/html/font-weight-bold-for-b-and-strong-expected.png: Ditto
* LayoutTests/platform/mac/fast/html/font-weight-bold-for-b-and-strong-expected.txt: Ditto
* LayoutTests/platform/win/fast/html/font-weight-bold-for-b-and-strong-expected.txt: Ditto
* LayoutTests/platform/glib/fast/html/font-weight-bold-for-b-and-strong-expected.txt: Ditto
* LayoutTests/fast/html/font-weight-bold-for-b-and-strong.html: Ditto

* LayoutTests/fast/html/font-weight-bolder-for-b-and-strong.html:
* LayoutTests/fast/html/font-weight-bolder-for-b-and-strong-expected.html:
* LayoutTests/platform/win/TestExpectations: Update due to test name change

Canonical link: <a href="https://commits.webkit.org/291855@main">https://commits.webkit.org/291855@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7c82dda7e57d3289956cf4fe05267e0ad9f98fc9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/94157 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/13744 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/3508 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/99168 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/44684 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/96207 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/14044 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/22174 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/71820 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29162 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/97159 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/10416 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/85041 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52161 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/10100 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/2700 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/44002 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/80333 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/2790 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/101210 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/21209 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/15443 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/80822 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/21461 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/81028 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/80204 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20009 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/24754 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/2118 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/14371 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/21193 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/26372 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/20880 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/24340 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/22621 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/fast/html/font-weight-bolder-for-b-and-strong.html:
Can we avoid render tree dumps as references? Can you simply make a copy of this file as ref that uses spans instead of b/strong with the correct font-weight?

Ahmad-S792 replied:
Can explore. Plus I know why it is asserting, didn't get time to get back to it. So will fix assert issue as well.

Ahmad-S792 replied:
@nt1m - Done - changed to reftest and fixed debug issue as well and added explanation in commit message.

nt1m reviewed (commented):
Looks like this is causing debug asserts.

I'm slightly concerned about potential fallout in mail or books but perhaps it's minimal?

Ahmad-S792 committed:
Sync `b` and `strong` to `font-weight: bolder` as per Web Specification

https://bugs.webkit.org/show_bug.cgi?id=289297
rdar://146458131

Reviewed by Tim Nguyen.

This patch aligns WebKit with Web Specification [1]:

[1] https://html.spec.whatwg.org/multipage/rendering.html#phrasing-content-3

Our current implementation had `font-weight: bold` for `strong` and `b` and
this patch fixes it to make it `font-weight: bolder` and align with other
browser engines as well.

It also updates local test similar to below Blink Commit [2]:

[2] https://chromium-review.googlesource.com/c/chromium/src/+/6269883

From this change, we start hitting `primitiveValue->isNumber()` assertion and
looking into `fontWeightValueIsBold` function, we were hitting this assertion because
in `switch` statement, we were not handling `CSSValueBolder`, so `bolder` was
hitting assertion, which is to ensure that it is `number` only.

* LayoutTests/TestExpectations:
* Source/WebCore/css/html.css:
(strong, b):
* Source/WebCore/editing/EditingStyle.cpp:
(fontWeightValueIsBold):

> Renamed `font-weight-bold-for-b-and-strong.html` to `font-weight-bolder-for-b-and-strong.html`:
* LayoutTests/platform/gtk/fast/html/font-weight-bold-for-b-and-strong-expected.png: Removed.
* LayoutTests/platform/ios/fast/html/font-weight-bold-for-b-and-strong-expected.txt: Ditto
* LayoutTests/platform/mac/fast/html/font-weight-bold-for-b-and-strong-expected.png: Ditto
* LayoutTests/platform/mac/fast/html/font-weight-bold-for-b-and-strong-expected.txt: Ditto
* LayoutTests/platform/win/fast/html/font-weight-bold-for-b-and-strong-expected.txt: Ditto
* LayoutTests/platform/glib/fast/html/font-weight-bold-for-b-and-strong-expected.txt: Ditto
* LayoutTests/fast/html/font-weight-bold-for-b-and-strong.html: Ditto

* LayoutTests/fast/html/font-weight-bolder-for-b-and-strong.html:
* LayoutTests/fast/html/font-weight-bolder-for-b-and-strong-expected.html:
* LayoutTests/platform/win/TestExpectations: Update due to test name change

Canonical link: https://commits.webkit.org/291855@main
